sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -148,14 +149,17 @@ trait Block extends JavaCode {
   }
 
   // Concatenates this block with other block.
-  def + (other: Block): Block
+  def + (other: Block): Block = other match {
+    case EmptyBlock => this
+    case _ => code"$this\n$other"
```

Comment:
why do we need `\n` here? It may be a single space as well in many cases or even nothing, IIUC

Reply:
The concatenation of two blocks needs a newline between them like following:

    [block1]
    [block2]

In embedding case, like code"$block1 ... $block2", no extra newlines are added.

Reply:
I see, thanks for your explanation